### PR TITLE
Finish: Part 2

### PR DESCRIPTION
**Pull request by lukman155**
In this PR,

 Add a column species of type string to the animals table
n1-after

 Insert data
n1

 Inside a transaction:
     Update the animals' table by setting the species column to digimon for all animals with a name ending in mon and update the animal's table by setting the species column to pokemon for all animals that don't have species already set.
n2

 Inside a transaction delete all records in the animals table, then roll back the transaction. After the rollback verify if all records in the animals table still exists.
n3
Rollback
n3-after

 Delete all animals born after Jan 1st, 2022.
Create a savepoint for the transaction.
n4

Update all animals' weight to be their weight multiplied by -1.
n4-negative

Rollback to the savepoint
Update all animals' weights that are negative to be their weight multiplied by -1.
n4-positive

 Write queries to answer the following questions:
How many animals are there?
n5

How many animals have never tried to escape?
n6

What is the average weight of animals?
n7

Who escapes the most, neutered or not neutered animals? Neutered animals
n8

What is the minimum and maximum weight of each type of animal?
n9

What is the average number of escape attempts per animal type of those born between 1990 and 2000?
n10

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -1,11 +1,67 @@
...
 SELECT * FROM animals
-SELECT * FROM animals WHERE neutered = 't';
\ No newline at end of file
+SELECT * FROM animals WHERE neutered = 't';
+
+/* Rollback practise */
+ BEGIN;
+ UPDATE animals SET species = 'unspecified';
+ ROLLBACK;
+
```
Comment (franclobo) [inline queries.sql:18]: - [ ] To verify that the changes were undone please add this line 👇🏽 

image

File: queries.sql [hunk 2]
```diff
@@ -1,11 +1,67 @@
...
+/* Add species data as transaction */
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon%';
+UPDATE animals SET species='pokemon' WHERE species IS NULL;
+COMMIT;
+
```
Comment (franclobo) [inline queries.sql]: - [ ] To verify that the changes persist after changes, please 🙏🏽 add this line 👇🏽 

image

File: queries.sql [hunk 3]
```diff
@@ -1,11 +1,67 @@
...
+ SELECT COUNT (*) FROM animals WHERE escape_attempts = 0;
+
+ /*What is the average weight of animals?*/
+
+  SELECT AVG(weight_kg) FROM animals;
+
+ /*Who escapes the most, neutered or not neutered animals?*/
+
+  SELECT neutered, SUM(escape_attempts) FROM animals GROUP BY neutered;
```
Comment (franclobo) [inline queries.sql:61]: - [ ] It could be better to use AVG or MAX instead of SUM to improve your query. 👍🏽

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @lukman155, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Correct Git flow ✔️ 
- Descriptive PR ✔️ 
- Correct data ✔️ 
- Correct queries ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] Please make the changes required in the schema file 👇🏽 

image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi @lukman155 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
gabriel_Igla


## Highlights
 - [ ] Good job making the changes requested by the previous code reviewer. ✅
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**